openlibrary/plugins/upstream/spamcheck.py

```diff
@@ -42,7 +43,7 @@ def is_spam(i=None):
     if i is None:
         i = web.input()
     text = str(dict(i)).lower()
-    return any(w.lower() in text for w in spamwords)
+    return any(re.search(w.lower(), text) for w in spamwords)
```

Comment:
Not a fan.  Are we editing spam words to contain regular expressions?  If not then I would not be supportive of this change.

If a programmer has a problem and they try to solve that problem with regular expressions, now they have two problems...  It is a joke but sometimes it can be not funny.  Let's reserve regex for places where it is really required.

Reply:
I tried to allude to the context in the issue without getting into too many details in public, but we already have multiple anti-spam bots running that each use different sets of regular expressions, and at least one of them already auto-bans accounts based on them.  The high-level plan discussed with mek is to unify all the regular expressions across the bots and the related admin tooling for consistency and to try to prevent the spam from showing up in the first place, before bots need to act.  The implementation is not carved in stone, so there's certainly room for feedback (thanks for your idea on the issue).